original_sdk/client.py

```diff
@@ -41,11 +46,15 @@ def set_http_session(self, session: requests.Session) -> None:
 
     def _parse_response(self, response: requests.Response) -> OriginalResponse:
         try:
-            parsed_result = json.loads(response.text) if response.text else {}
+            parsed_result = response.json() if response.text else {}
         except ValueError:
-            raise OriginalAPIException(response.text, response.status_code)
-        if response.status_code >= 399:
-            raise OriginalAPIException(response.text, response.status_code)
+            raise ClientError(
+                message="Invalid JSON received",
+                status=response.status_code,
+                data=response.text,
+            )
+        if is_error_status_code(response.status_code):
```

Comment:
we should not combine both?

Reply:
The types are slightly different for instance

- requests.Response uses `status`
- aiohttp.ClientResponse uses `status_code` 

Plus we need to await the async responses

Reply:
@mgom updated to reuse as much as possible, but the sync/async differences means we need to keep some parts separate (getting the response details + `await`)